src/ray/raylet/node_manager.h

```diff
@@ -641,8 +641,9 @@ class NodeManager : public rpc::NodeManagerServiceHandler {
   /// The new resource scheduler for direct task calls.
   std::shared_ptr<ClusterResourceScheduler> new_resource_scheduler_;
   /// Map of leased workers to their current resource usage.
-  std::unordered_map<WorkerID, std::unordered_map<std::string, double>>
-      leased_worker_resources_;
+  // XXX std::unordered_map<WorkerID, std::unordered_map<std::string, double>>
+  // XXX    leased_worker_resources_;
+  std::unordered_map<WorkerID, ResourceSet> leased_worker_resources_;
```

Comment:
Can this be tracked inside the worker class itself (I thought it was already?)

Reply:
This variable was already in the previous code. I just changed the type, so it returns the resource set instead of the map of resource names to quantities, since I need resource sets in the code to  track resource ids.

This being said, we need to remove it. The issue right now that the task_id is not associated to its worker in the new code. We can talk more about this in person and fix in the next sprint.